activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -514,10 +514,6 @@ def check_constraint(expression, **options)
       def timestamps(**options)
         options[:null] = false if options[:null].nil?
 
-        if !options.key?(:precision) && @conn.supports_datetime_with_precision?
-          options[:precision] = 6
-        end
```

Comment:
This change isn't so much an optimization as it is clean up.

By not setting it here, we'll end up running the check twice (once for each column). However, I think that is better, as this is not a performance critical path, so the improvement to code maintainability (by not having the same logic in multiple places) is worth it.